### PR TITLE
Bump bcftools to v1.18

### PR DESCRIPTION
**Pull request by nvnieuwk**
## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

**Review by FriederikeHanssen — APPROVED** [review]

**Review by mashehu — APPROVED** [review]

**Review by mashehu — APPROVED** [review]
some copy-paste mistakes

File: modules/nf-core/bcftools/call/tests/main.nf.test [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+            process {
+                """
+                input[0] = [
+                    [ id:'out', single_end:false ], // meta map
+                    file(params.test_data['sarscov2']['illumina']['test_vcf_gz'], checkIfExists: true),
+                    file(params.test_data['sarscov2']['illumina']['test_vcf_gz_tbi'], checkIfExists: true)
+                ]
+                input[1] = []
+                input[2] = []
+                input[3] = []
```
[the remark this answers is not in the document]
Comment (nvnieuwk) [reply modules/nf-core/bcftools/call/tests/main.nf.test:26]: These inputs are present in the `main.nf`
```
process BCFTOOLS_CALL {
    tag "$meta.id"
    label 'process_medium'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/bcftools:1.18--h8b25389_0':
        'biocontainers/bcftools:1.18--h8b25389_0' }"

    input:
    tuple val(meta), path(vcf), path(index)
    path(regions)
    path(targets)
    path(samples)
```

File: modules/nf-core/bcftools/call/tests/main.nf.test [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert snapshot(
+                    file(process.out.vcf[0][1]).name,
```
[the remark this answers is not in the document]
Comment (nvnieuwk) [reply modules/nf-core/bcftools/call/tests/main.nf.test:94]: This is a stub test (not present in the previous tests). The linting fails if an md5sum is calculated of empty files, that's why I only get the name in stub tests

File: modules/nf-core/bcftools/query/tests/main.nf.test [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+            process {
+                """
+                input[0] = [ 
+                    [ id:'out' ], // meta map
+                    file(params.test_data['sarscov2']['illumina']['test_vcf_gz'], checkIfExists: true),
+                    file(params.test_data['sarscov2']['illumina']['test_vcf_gz_tbi'], checkIfExists: true)
+                ]
+                input[1] = []
+                input[2] = []
+                input[3] = []
```
[the remark this answers is not in the document]
Comment (nvnieuwk) [reply modules/nf-core/bcftools/query/tests/main.nf.test:26]: These are also inputs of the modules:
```
process BCFTOOLS_QUERY {
    tag "$meta.id"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/bcftools:1.18--h8b25389_0':
        'biocontainers/bcftools:1.18--h8b25389_0' }"

    input:
    tuple val(meta), path(vcf), path(tbi)
    path regions
    path targets
    path samples
```

File: modules/nf-core/bcftools/call/tests/main.nf.test (hunk 1 above)
[the remark this answers is not in the document]
Comment (mashehu) [reply modules/nf-core/bcftools/call/tests/main.nf.test:26]: oh, sorry 👍🏻

**Review by johnoooh — APPROVED** [review]

**Pull request merged**